### PR TITLE
feat: support gitee catalog

### PR DESCRIPTION
**Pull request by alexcodelf**
**Problem:**

Template Catalog is not supported.

**Solution:**

Add gitee client to support gitee catalog. This is partial Pull Request to address #1736 we need support catalog server mirror [walrus-catalog](https://github.com/walrus-catalog) to gitee.

**Additional**
cc @hibig UI need a new catalog type `Gitee`, one thing to note is that template icons cannot be displayed from gitee, as gitee has implemented measures to prevent hot-linking of images, all image link will return 403. The possible measure is to set `<meta name="referrer" content="no-referrer">`

**Related Issue:**
#1736

### CODE REVIEW

**Review by gitlawr — DISMISSED** [review]
LGTM

File: pkg/server/runner.go [hunk 1]
```diff
@@ -136,6 +139,20 @@ func (r *Server) Flags(cmd *cli.Command) {
...
 			Value:       r.EnableTls,
 		},
+		&cli.StringFlag{
+			Name:        "builtin-catalog-provider",
+			Usage:       "Builtin catalog provider, Github or Gitee.",
+			Destination: &r.BuiltinCatalogProvider,
+			Value:       r.BuiltinCatalogProvider,
+			Action: func(c *cli.Context, s string) error {
+				switch s {
+				case types.GitDriverGitee, types.GitDriverGithub:
```
Comment (gitlawr) [inline pkg/server/runner.go:149]: Should we make it case insensitive in cli args?

Comment (alexcodelf) [reply pkg/server/runner.go]: Updated, it's value is case-insensitive now.

**Review by gitlawr — APPROVED** [review]
LGTM

**Review by aiwantaozi — APPROVED** [review]
LGTM

**Pull request merged**